### DISCUSSION TITLE
Last update : System Variable un CCU now appears as binary_sensor

### DISCUSSION THREAD

**Discussion by cy-bertrand** [SukramJ/aiohomematic#754] [Q&A] [answered]
Hello,

Since last update of the integration, some entities have disapeared from my HA.
Precisely what i've seen is that all my **system variables** (binary ones) are now integrated as **binary_sensor** instead of **sensor**.

The problem I've got is that I can not change the value of a binary_sensor in HA but a sensor can be switched on/off.
So now, let's say it is a read only value.

Aa-s my CCU is reading theses variables for several programs, it is now impossible to control these programs anymore through HA :-(

Anyway to recover the variable as sensor ?

Many thanks,

Bertrand.

**Comment by SukramJ** [comment]
I read your text several times, but i have no clue what you are talking about!
It's also possible to write it i german if it helps.

Please describe only one case/problem with one sysvar. Generalization with last, one, some, all are not helpful

## Add information like requested in an issue template.

Please answer all questions as good as possible.

Language: as you like -> english/DEUTSCH 

Make sure you have read the documentation at https://github.com/danielperna84/custom_homematic#custom_homematic.

Make sure you have read the release notes at https://github.com/danielperna84/custom_homematic/releases. 

Make sure you are running the latest version of the custom_component (and Home Assistant) before reporting an issue.

ENSURE that your Backend (CCU/Homegear/...) is working as expected and connected devices are controllable.


**custom_component version (if applicable):**
1.x.x

**known last working custom_component version (if applicable):**
1.x.x

**Home Assistant:**
- Version: 2022.x.x
- Installation method (HAOS, supervised, core, venv, Docker): 

**CCU:**
- Model (CCU2/3, Raspi3/4, NUC, ProxmoxVE, HA-Addon ...):
- OS (Original, RaspberryMatic, debmatic ...):
- Version: 

**Problem-relevant configuration:**

Do you use tls?
Do you use callback?
Do you use username and password?
Which interfaces do you use (hmip/bidcos/wired)?

**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Screenshots (if applicable)**
- CCU firewall in CCU
- device config in CCU
- integrations view in HA
- device view in HA
- enity view in HA

**Relevant HA log entries (press full log) VERY IMPORTANT**
```


```

**Diagnostic Information (available on every device)**
```


```

**Expected behavior**
A clear and concise description of what you expected to happen.

**Additional context**
Add any other context about the problem here.

**Comment by Baxxy13** [comment, marked as answer]
@cy-bertrand 
To use boolean SysVars as "switches" mark them with `hahm` in their description (WebUI).
And read the [README](https://github.com/danielperna84/custom_homematic/blob/devel/README.md#system-variables).

**Reply by cy-bertrand** [reply]
Great, perfect !

(note that after adding `hahm` in the description of my sysvar on the CCU, it is automatically integrated as sensor in HA with a "Sv_" in the name.  And theses new entities are desactivated by the integration by default)

Strange that it was working without needing the marker before the last serie of updates of the integration.

Many thanks @Baxxy13 
Sorry @SukramJ for the difficulties to be understable ;-) Baxxy gave me the answer.

**Reply by SukramJ** [reply]
> Strange that it was working without needing the marker before the last serie of updates of the integration.

The marker is required since this feature was introduced.
